bundles/org.openhab.automation.jsscripting/src/main/resources/node_modules/@jsscripting-globals.js

```diff
@@ -0,0 +1,172 @@
+                } catch (_) {
+                    return '[Circular]';
+                }
+            // falls through
+            default:
+                return x;
+        }
+    });
+    for (var x = args[i]; i < len; x = args[++i]) {
+        if (x === null || (typeof x !== 'object' && typeof x !== 'symbol')) {
+            str += ' ' + x;
+        } else {
+            str += ' ' + stringify(x);
+        }
+    }
+    return str;
+}
+
+const counters = {};
+const timers = {};
+
+const console = {
+    'assert': function (expression, message) {
+        if (!expression) {
+            log.error(message);
+        }
+    },
+
+    count: function (label) {
+        let counter;
+
+        if (label) {
+            if (counters.hasOwnProperty(label)) {
+                counter = counters[label];
+            } else {
+                counter = 0;
+            }
+
+            // update
+            counters[label] = ++counter;
+            log.debug(format.apply(null, [label + ':', counter]));
+        }
+    },
+
+    debug: function () {
+        log.debug(format.apply(null, arguments));
+    },
+
+    info: function () {
+        log.info(format.apply(null, arguments));
+    },
+
+    log: function () {
+        log.info(format.apply(null, arguments));
+    },
+
+    warn: function () {
+        log.warn(format.apply(null, arguments));
+    },
+
+    error: function () {
+        log.error(format.apply(null, arguments));
+    },
+
+    trace: function (e) {
+        if (Java.isJavaObject(e)) {
+            log.trace(e.getLocalizedMessage(), e);
+        } else {
+            if (e.stack) {
+                log.trace(e.stack);
+            } else {
+                if (e.message) {
+                    log.trace(format.apply(null, [(e.name || 'Error') + ':', e.message]));
+                } else {
+                    log.trace((e.name || 'Error'));
+                }
+            }
+        }
+    },
+
+    time: function (label) {
+        if (label) {
+            timers[label] = System.currentTimeMillis();
+        }
+    },
+    timeEnd: function (label) {
+        if (label) {
+            const now = System.currentTimeMillis();
+            if (timers.hasOwnProperty(label)) {
+                log.info(format.apply(null, [label + ':', (now - timers[label]) + 'ms']));
+                delete timers[label];
+            } else {
+                log.info(format.apply(null, [label + ':', '<no timer>']));
+            }
+        }
+    }
+};
+
+function setTimeout(cb, delay) {
+    return ScriptExecution.createTimer(
+        ZonedDateTime.now().plusNanos(delay * 1000000),
+        cb
+    );
+}
+
+function clearTimeout(timer) {
+    if (timer.isActive() && !timer.isRunning()) {
+        timer.cancel();
+    }
+}
+globalThis.console = console;
+globalThis.setTimeout = setTimeout;
+globalThis.clearTimeout = clearTimeout;
+globalThis.global = globalThis;
+globalThis.process = { env: { NODE_ENV: '' } };
```

Comment:
When I added this in my scripts, this was to prevent npm libraries from crashing when they access `process.env.NODE_ENV`. For which this is enough. But most often this is used to check if it equals `'production'`, in which case libraries often emit less/no debug logging, type checks, or do other shortcuts to optimize performance. 

So there is a tradeoff here basically between DX and performance. I would tend to favor DX in our case, but should we want to prioritize perf, then this could be changed to:

```suggestion
globalThis.process = { env: { NODE_ENV: 'production' } };
```